src/python_framework_v02/troute/nhd_io.py

```diff
@@ -711,28 +712,48 @@ def get_usgs_from_time_slices_folder(
     last_obs_NaN = {"obs":NaN, "time":NaT}  # No valid recent observation
     ```
     """
-    
+
     # ---- Laugh testing ------
     # scren-out erroneous qc flags
     usgs_qual_df = usgs_qual_df.mask(usgs_qual_df < 0, np.nan)
     usgs_qual_df = usgs_qual_df.mask(usgs_qual_df > 1, np.nan)
-    
+
     # screen-out poor quality flow observations
     qc_trehsh = 1
     usgs_df = usgs_df.mask(usgs_qual_df < qc_trehsh, np.nan)
-    
+
     # screen-out erroneous flow observations
     usgs_df = usgs_df.mask(usgs_df < 0, np.nan)
-        
+
     # ---- Interpolate USGS observations to time discretization of the simulation ----
-    usgs_df.columns = pd.to_datetime(usgs_df.columns, format = "%Y-%m-%d_%H:%M:00")
-    max_fill = 5
-    usgs_df = usgs_df.interpolate(method='index', axis = 1, limit = max_fill, limit_direction = 'forward')
+    usgs_df_T = usgs_df.transpose()
+    usgs_df_T.index = pd.to_datetime(usgs_df_T.index, format = "%Y-%m-%d_%H:%M:%S")
 
-    usgs_df.drop(usgs_df[usgs_df.iloc[:, 0] == -999999.000000].index, inplace=True)
-    usgs_qual_df.drop(usgs_df[usgs_df.iloc[:, 0] == -999999.000000].index, inplace=True)
+    max_fill_1min = 59
+    # max_fill_1min = 14
+    # max_fill = 5
+    """
+    Note: The max_fill is applied when the series is being considered at a 1 minute interval
+    14 minutes ensures no over-interpolation with 15-minute gage records, but creates
+    square-wave signals at gages reporting hourly.
 
-    return usgs_df
+    TODO: Add reporting interval information to the gage preprocessing (timeslice generation)
+    """
+
+    usgs_df_T = usgs_df_T.resample('min').interpolate(limit = max_fill_1min, limit_direction = 'both').resample('5min').asfreq()
+    usgs_df_T_T = usgs_df_T.transpose()
```

Comment:
I left this step in here to help with clarity. If we are satisfied, we could concatenate some of these options.